core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCTemplate.scala

```diff
@@ -0,0 +1,364 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.ContractOraclePair._
+import org.bitcoins.core.protocol.dlc.DLCTemplate._
+import org.bitcoins.core.protocol.tlv._
+
+sealed trait DLCTemplate {
+  def totalCollateral: CurrencyUnit
+
+  def oracles: Vector[OracleAnnouncementTLV]
+
+  def oracleThreshold: Int
+
+  def contractDescriptor: ContractDescriptor
```

Comment:
I don't understand how we can have a `ContractDescriptor` if we are building a DLC? 

In my mind, the `DLCTemplate` should maybe be renamed to `ContractDescriptorBuilder` as it is taking input that will eventually be used to build a contract descriptor. 

We then use that contract descriptor inside of something like `DLCTemplate` to add in things like the oracles.

So I guess what I am saying is this stuff seems to make sense, but there is a level below this I think we need to add (building the `ContractDescriptor`)